stablehlo/reference/ProcessGrid.cpp

```diff
@@ -154,6 +154,25 @@ void ProcessGrid::outfeed(ArrayRef<Tensor> inputs) {
   outfeed_.emplace(inputs);
 }
 
+SmallVector<Tensor> ProcessGrid::recv(ChannelId channelId,
+                                      ProcessId processId) {
+  {
+    std::lock_guard<std::mutex> lock(sendRecvLocks_[channelId]);
```

Comment:
This will probably have the same issue we hit in https://github.com/openxla/stablehlo/pull/1737 right?

We could make a new datatype `MutexMap` which wraps this map and has a synchronized getter method.

Reply:
Actually wait.. do we know the number of mutexes we will need at creation time? We can instead use a vector of mutexes and index by channel id

Reply:
Thanks for following up in this PR! I haven't made the changes following up from #1737 yet.

> do we know the number of mutexes we will need at creation time?

Technically possible (I think), but not feasible. We can have multiple send/recv (or any other distribution ops) with different `channelId`s within a single process. This would mean going through op by op to obtain the set of used `channelId`s.

> We can instead use a vector of mutexes and index by channel id

This won't work since `channelId` can be arbitrary (i.e. does not have to increment 0, 1, 2, etc. unlike `channelType` which is enumerated), and map would be a better fit for this reason.

Reply:
Updated.